### ISSUE TITLE
Implement the launcher screen

### ISSUE THREAD

**Issue by Wohlstand** [Wohlstand/TheXTech#40]
Some time ago, for the purposes of the SMR project, I developed the launcher as a separate application: https://github.com/Core-LOVE/nsmbx-launcher. Its purpose is to give the ability to let user tune some gameplay settings that required to be set before starting the game itself (also, it can be used to re-implement the password system for the ASMBXT: Prelude To The Stupid project). Right now the Android version has a simple launcher that works. The launcher can be implemented as the in-engine screen that will be shown when no special command-line arguments were specified to allow the user to tune the game settings. It should be controllable via both mouse as well as a touch-screen, and by keyboard by users who don't want or can't use the mouse for the settings menu navigation.

**Comment by Wohlstand** [comment]
> Hmm, would it allow the launching of an editor, such as the PGE Editor?

As devkit is a separate thing, then, it's would be an option for the case if devkit going to be included with the game set. But when devkit is absent, just don't show that menu item.

**Comment by Wohlstand** [comment]
> Would it have Gamepad support for devices that may only have a gamepad?

Yes, I guess it should be using the GameController API that allows automapping of known gamepads, however, when the gamepad device is unknown, it would fall back into the SDL Joystick API that will require the manual mapping setup of it like in older TheXTech versions and at Moondust right now (soon I gonna backport the most of updates I done here into Moondust Engine back)

**Comment by ChristianSilvermoon** [comment]
> > Would it have Gamepad support for devices that may only have a gamepad?
> 
> Yes, I guess it should be using the GameController API that allows automapping of known gamepads, however, when the gamepad device is unknown, it would fall back into the SDL Joystick API that will require the manual mapping setup of it like in older TheXTech versions and at Moondust right now (soon I gonna backport the most of updates I done here into Moondust Engine back). It's already working with the Retroid Pocked device, it gets automatically toggled and mapped, so, simply start the game and play.

Oh *very* nice ^^

**Comment by ds-sloth** [comment]
Why do we need this instead of improving the settings menu to include all of these options? In other words, which options need to be set by the command line and thus require a new launcher program to set? Maintainability across all devices would be twice as difficult if we need to support all input APIs twice.

**Comment by ChristianSilvermoon** [comment]
> Why do we need this instead of improving the settings menu to include all of these options? In other words, which options need to be set by the command line and thus require a new launcher program to set? Maintainability across all devices would be twice as difficult if we need to support all input APIs twice.

I'm in favour of having things in the settings menu, but not at the expensive a powerful and extensive set of Command Line arguments, personally.

**Comment by ds-sloth** [comment]
We have continued to grow and improve the in-game menus and I wonder if this is now out-of-scope for the project.

**Comment by LoveBodhi** [comment]
Is this seems Riivolution-like options?

**Comment by ChristianSilvermoon** [comment]
> We have continued to grow and improve the in-game menus and I wonder if this is now out-of-scope for the project.

A launcher seems like it should be a separate project to me.

TheXTech has a robust set of command-line arguements that enables the easy creation of launchers for those who'd prefer one or want to use one... but I think that the base game itself should be designed with the intention of it not being necessary to have one.

So I agree, and feel that a launcher itself is out of scope and that the functionality one presents should be accessible via in-game menus instead.

**Comment by ds-sloth** [comment]
Krissy explained it better than I can.
